packages/dd-trace/test/appsec/iast/utils.js

```diff
@@ -153,10 +100,107 @@ function copyFileToTmp (src) {
   return dest
 }
 
+function prepareTestServerForIast (description, tests) {
+  describe(description, () => {
+    const config = {}
+    let http
+    let listener
+    let appListener
+    let app
+
+    before(() => {
+      return getPort().then(newPort => {
+        config.port = newPort
+      })
+    })
+
+    before(() => {
+      listener = (req, res) => {
+        const appResult = app && app(req, res)
+        if (appResult && typeof appResult.then === 'function') {
+          appResult.then(() => {
+            res.writeHead(200)
+            res.end()
+          })
+        } else {
+          res.writeHead(200)
+          res.end()
+        }
+      }
+    })
+
+    before(() => {
+      return agent.load('http', undefined, { flushInterval: 1 })
+        .then(() => {
+          http = require('http')
```

Comment:
There is no need to call done here? It is resolved by the Promise?

Reply:
exactly, if `before`, `beforeEach`... returns a promise, it will wait until it is finished.